src/maps/google-maps-provider.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @typedef {import('../../types/src/maps/maps.service').MapsAPIProvider} MapsAPIProvider
+ * @typedef {import('../../types/src/maps/maps.service').GetDirectionInfo} GetDirectionInfo
+ * @typedef {import('../../types/src/maps/google-maps-provider').GoogleDirectionResponse} GoogleDirectionResponse
+ * @typedef {import('../../types/src/maps/maps.service').MapsAPIProviderDeps} MapsAPIProviderDeps
+ * @typedef {import('../../types/src/maps/maps.service').DirectionInfoParams} DirectionInfoParams
+ */
+
+import { partial } from '@oldbros/shiftjs';
+import { makeHttpRequest } from '../utils/http-request.utils.js';
+
+const API_URL = 'https://maps.googleapis.com/maps/api/directions/json';
+const GOOGLE_API_KEY = process.env.GOOGLE_API_KEY;
+/**
+ * @param {MapsAPIProviderDeps} deps
+ * @param {DirectionInfoParams} params
+ */
+export const getDirectionInfo = async (deps, { departurePoint, destinationPoint }) => {
+  const { logger } = deps;
+  try {
+    /** @type {GoogleDirectionResponse} */
+    // eslint-disable-next-line max-len
+    const response = await makeHttpRequest(`${API_URL}?origin=${departurePoint.long}, ${departurePoint.lat}&destination=${destinationPoint.long}, ${destinationPoint.lat}&key=${GOOGLE_API_KEY}`);
```

Comment:
It is better to have simple function wich prepares a URL, similar as you did for Headers in JWT